migrate2rocky/migrate2rocky.sh

```diff
@@ -350,13 +350,37 @@ collect_system_info () {
     # Check the efi mount first, so we can bail before wasting time on all these
     # other checks if it's not there.
     if [[ $update_efi ]]; then
-	local efi_mount
-	declare -g efi_disk efi_partition
+	local efi_mount kname
+	declare -g -a efi_disk efi_partition
 	efi_mount=$(findmnt --mountpoint /boot/efi --output SOURCE \
 	    --noheadings) ||
 	    exit_message "Can't find EFI mount.  No EFI  boot detected."
-	efi_disk=/dev/$(lsblk -no pkname "$efi_mount")
-	efi_partition=$(<"/sys/block/${efi_disk##*/}/${efi_mount##*/}/partition")
+	kname=$(lsblk -dno kname "$efi_mount")
+	efi_disk=$(lsblk -dno pkname "/dev/$kname")
```

Comment:
This act of assigning to efi_disk here creates a first array entry (sets `efi_disk[0]=''`) when not a vanilla partition, so the loop at 820 fails on the first index due to an empty disk name. (As `$efi_disk` ends up being `['' , 'foo' , 'bar']` rather than `['foo' , 'bar']` as intended.)

Doing this works (on my md-raid setup), and should also work on vanilla:

```diff
--- migrate2rocky_68.sh	2021-07-06 16:50:40.000000000 -0500
+++ migrate2rocky_fixed.sh	2021-07-06 16:50:31.000000000 -0500
@@ -356,9 +356,10 @@
 	    --noheadings) ||
 	    exit_message "Can't find EFI mount.  No EFI  boot detected."
 	kname=$(lsblk -dno kname "$efi_mount")
-	efi_disk=$(lsblk -dno pkname "/dev/$kname")
+	efi_simple=$(lsblk -dno pkname "/dev/$kname")

-	if [[ $efi_disk ]]; then
+	if [[ $efi_simple ]]; then
+	    efi_disk=$efi_simple
 	    efi_partition=$(<"/sys/block/$efi_disk/$kname/partition")
 	else
 	    # This is likely an md-raid or other type of virtual disk, we need```

Reply:
Thanks for the feedback.  Can you try again with the newest commit?

Reply:
Runs without errors; `efibootmgr` reports the new entries and boots as expected. Thanks!